server/tests/endpoints/certification/test_certification.py

```diff
@@ -110,6 +171,8 @@ def test_hardware_mismatch(generator: DataGenerator, test_client: TestClient):
 
     assert response.status_code == 200
     assert response.json() == {"status": "Not Seen"}
+    assert "Hardware cannot be found" in caplog.text
+    assert "bios version: 1.0.15" in caplog.text
```

Comment:
I see that the logs are being used to test that the reason hardware is "Not Seen" is what's expected of this test. It's great that we're testing this. But I suggest that instead of depending on logs, we test it directly. This can be done by refactoring the logic that checks if hardware is certified in it's own function. This function would be used by the controller. It can return not only the status but also the reason. Then we can write tests for that function to make sure the reason matches what we expected.

Reply:
Matias asked to check that if there is a hardware mismatch, we log this information